static/js/src/publisher/listing-page.js

```diff
@@ -1,24 +1,159 @@
-import { initMultiselect } from "./components/multiselect";
-
-function toggleModal(modal) {
-  if (
-    modal &&
-    modal.classList.contains("p-tooltip__modal") &&
-    window.innerWidth < 772
-  ) {
-    modal.classList.toggle("is-open");
+// https://gist.github.com/dperini/729294
+// Luke 07-06-2018 made the protocol optional
+const URL_REGEXP = /^(?:(?:https?|ftp):\/\/)?(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,}))\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/i;
+// Luke 07-06-2018 rather then looking for a mailto, look for 1 @ and at least 1 .
+const MAILTO_REGEXP = /[^@]+@[^@]+\.[^@]+/;
+
+class ListingForm {
+  constructor(selector) {
+    this.formEl = document.querySelector(selector);
+
+    if (this.formEl) {
+      this.validation = {};
+      this.prefixableFields = ["contact"];
+      this.validateInputs = Array.from(
+        this.formEl.querySelectorAll("input,textarea")
+      );
+
+      this.init();
+    } else {
+      throw new Error(`${selector} is not a valid element`);
+    }
   }
-}
 
-function initTooltips() {
-  document.addEventListener("click", function (event) {
-    var targetControls = event.target.getAttribute("aria-controls");
-    if (targetControls) {
-      toggleModal(document.getElementById(targetControls));
+  init() {
+    const that = this;
+
+    this.validateInputs.forEach((input) => {
+      const inputValidation = { isValid: true };
+
+      if (input.maxLength > 0) {
+        // save max length, but remove it from input so more chars can be entered
+        inputValidation.maxLength = input.maxLength;
+
+        // prepare counter element to show how many chars need to be removed
+        const counter = document.createElement("p");
+        counter.className = "p-form-help-text";
+        inputValidation.counterEl = counter;
+        input.parentNode.appendChild(counter);
+      }
+
+      if (input.required) {
+        inputValidation.required = true;
+      }
+
+      if (input.type === "url") {
+        inputValidation.url = true;
+      }
+
+      // allow mailto: addresses for contact field
+      if (input.name === "contact") {
+        inputValidation.mailto = true;
+      }
+
+      that.validation[input.name] = inputValidation;
+    });
```

Comment:
I think if you add `.bind(this)` to the end of the `forEach` function you won't have to do `that = this`

Reply:
oh, yes. thanks